app/forms/feedback_form.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+class FeedbackForm
+  include ActiveModel::Model
+
+  class WrongQuestionFormatError < StandardError; end;
+
+  class << self
+    def model_name
+      ActiveModel::Name.new(self, nil, 'Form')
+    end
+
+    private
+
+    def title(title_name)
+      class_eval %Q{
+        def title
+          "#{title_name}"
+        end
+      }
+    end
+
+    def questions(qns = [])
+      qns.each do |qn|
+        raise(WrongQuestionFormatError, "Question must contain key") unless qn.include?(:key)
+        raise(WrongQuestionFormatError, "Question must contain question") unless qn.include?(:question)
+
+        class_eval %Q{
+          attr_accessor :#{qn[:key]}
+          #{ qn[:required] == true ? "validates :#{qn[:key]}, presence: true" : "" }
+        }
+      end
+
+      class_eval %Q{
+        def questions
+          #{qns}
+        end
+      }
+    end
+  end
+
+  attr_accessor :form_type
```

Comment:
Should this be replaced by just inlining L:47 in a method named `form_type`? Or at least be made read-only? 🙂

Reply:
good catch. it should be ready only!